docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
+
+To determine if the Exclusion List is going to be modified by a single `Chapter` or by an entire `Module`, `ExcludeCommand` implements an operation each for `exclude more` and `exclude less`.
+* For `exclude more`:
+    * `ExcludeCommand#addingToExclusion()` - calls `ExcludeCommand#addChapterToExclusion()` if user inputs "chapter", `ExcludeCommand#addModuleToExclusion()` if user inputs "module"
+* For `exclude less`:
+    * `ExcludeCommand#addingToExclusion()` - calls `ExcludeCommand#removeChapterFromExclusion()` if user inputs "chapter", `ExcludeCommand#removeModuleFromExclusion()` if user inputs "module"
+
+To load and store the Exclusion List, a Exclusion File is created and maintained using these two methods from the `Storage` Class:
+* `Storage#loadExclusionFile()` - Reads the contents of the Exclusion File, parses it into the Exclusion List, stored as a `ArrayList<String>`, and returns it.
+* `Storage#updateExclusionFile()` - Writes the `ArrayList<String>` Exclusion List into the Exclusion File.
+
+The `ArrayList<String>` Exclusion List is modified using four pairs of commands in both `ExcludeCommand` and `Storage`:
+
+* Excluding a Chapter from Scheduling
+    * `ExcludeCommand#addChapterToExclusion()` - gets the name of the `Chapter` to be excluded and the name of the `Module` it belongs to, and calls `Storage#appendChapterToExclusionFile()`
+    * `Storage#appendChapterToExclusionFile()` - appends the target `Chapter` to the Exclusion File if the target `Chapter` exists and is not already in the Exclusion File
+* Excluding a Module from Scheduling
+    * `ExcludeCommand#addModuleToExclusion()` - gets the name of the `Module` to be excluded, and calls `Storage#appendModuleToExclusionFile()`
+    * `Storage#appendModuleToExclusionFile()` - appends every `Chapter` of the target `Module` not already in the Exclusion File to it if the target `Module` exists
+* Including a Chapter from Scheduling
+    * `ExcludeCommand#removeChapterFromExclusion()` - gets the name of the `Chapter` to be included and the name of the `Module` it belongs to, and calls `Storage#removeChapterFromExclusionFile()`
+    * `Storage#removeChapterFromExclusionFile()` - removes the target `Chapter` from the Exclusion File if the target `Chapter` is in the Exclusion File
+* Including a Module from Scheduling
+    * `ExcludeCommand#removeModuleFromExclusion()`- gets the name of the `Module` to be included, and calls `Storage#removeModuleFromExclusionFile()`
+    * `Storage#removeModuleFromExclusionFile()` - removes every `Chapter` of the target `Module` that is in the Exclusion File
+
+The following sequence Diagrams illustrates how the Exclusion Process is executed:
+
+![](images/ExcludeCommand.png)
+
+![](images/addModuleToExclusion.png)
+
+![](images/addChapterToExclusion.png)
+
+![](images/removeModuleFromExclusion.png)
+
+![](images/removeChapterFromExclusion.png)
+<br>
```

Comment:
Perhaps the explanation/visuals used here is a little repetitive? Consider further splitting it into smaller sub-sections, or shorten the explanation when explaining the subsequent commands that are highly similar to previous ones.